pkgs/development/interpreters/octave/wrapper.nix

```diff
@@ -0,0 +1,119 @@
+{ stdenv, octave, buildEnv
+, makeWrapper, texinfo
+, octavePackages
+, extraLibs ? []
+, extraOutputsToInstall ? []
+, postBuild ? ""
+, ignoreCollisions ? false
+}:
+
+# Create an octave executable that knows about additional packages
+let
+  env = let
+    paths = extraLibs ++ [ octave ];
+    octavePath = "${octave}";
+    octaveExecutable = "${placeholder "out"}/bin/octave-cli";
+  in (buildEnv {
```

Comment:
You can get rid of the whole `let in` thing, and use all of these vars inside the arguments to `buildEnv`. Use `${octave}` instead of `${octavePath}`.

Reply:
See https://github.com/NixOS/nixpkgs/pull/108562/commits/12d41c4d14ab005b5cb07c5ca62b2a50b5ce5305